tool/tsh/tsh_test.go

```diff
@@ -411,3 +411,61 @@ func TestFormatConnectCommand(t *testing.T) {
 		})
 	}
 }
+
+// TestReadClusterFlag tests that cluster environment flag is reads in correctly.
+func TestReadClusterFlag(t *testing.T) {
+	var tests = []struct {
+		desc          string
+		inCLIConf     CLIConf
+		inSiteName    string
+		inClusterName string
+		outSiteName   string
+	}{
+		{
+			desc:          "nothing set",
+			inCLIConf:     CLIConf{},
+			inSiteName:    "",
+			inClusterName: "",
+			outSiteName:   "",
+		},
+		{
+			desc:          "TELEPORT_SITE set",
+			inCLIConf:     CLIConf{},
+			inSiteName:    "a.example.com",
+			inClusterName: "",
+			outSiteName:   "a.example.com",
+		},
+		{
+			desc:          "TELEPORT_CLUSTER set",
+			inCLIConf:     CLIConf{},
+			inSiteName:    "",
+			inClusterName: "b.example.com",
+			outSiteName:   "b.example.com",
+		},
+		{
+			desc:          "TELEPORT_SITE and TELEPORT_CLUSTER set, prefer TELEPORT_CLUSTER",
+			inCLIConf:     CLIConf{},
+			inSiteName:    "c.example.com",
+			inClusterName: "d.example.com",
+			outSiteName:   "d.example.com",
+		},
+		{
+			desc: "TELEPORT_SITE and TELEPORT_CLUSTER and CLI flag is set, prefer CLI",
+			inCLIConf: CLIConf{
+				SiteName: "e.example.com",
+			},
+			inSiteName:    "f.example.com",
+			inClusterName: "g.example.com",
+			outSiteName:   "e.example.com",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			os.Setenv(siteEnvVar, tt.inSiteName)
+			os.Setenv(clusterEnvVar, tt.inClusterName)
```

Comment:
I know this is probably fine at this point, but if other tests read the environment and are executed in parallel with this, there will be unexpected interference. Also, the environment needs to be cleaned up so I'd have `t.Cleanup` remove the environment variables at the end.
To remediate the first issue, readCusterFlag could take a replacement of the environment reader function. Alternatively, all other cases where the envvars are implicitly consumed should be audited and the use replaced with test values  which is subjectively more complex. Third option is running this test individually to explicitly avoid interference.